components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java

```diff
@@ -534,17 +535,41 @@ public Response throttlingPolicySearch(String query, MessageContext messageConte
             throws APIManagementException {
         ThrottlePolicyDetailsListDTO resultListDTO = new ThrottlePolicyDetailsListDTO();
         String policyType;
-        Map<String, String> filters;
+        String policyName;
+        Map<String, String> filters = new HashMap<>();
+
+        /**
+         * TODO: Add support to filter by name
+         */
         if (query == null) {
             query = "type:" + ALL_TYPES;
-        } else if (query.toLowerCase().indexOf("type:") != 0) {
+        } else if (query.toLowerCase().indexOf("type:") < 0) {
             String errorMessage = "Invalid query format";
             RestApiUtil.handleInternalServerError(errorMessage, log);
         }
+
         log.debug("Extracting query info...");
-        filters = Splitter.on(" ").withKeyValueSeparator(":").split(query);
+
+        try{
```

Comment:
Shall we fix this formatting issue?

Reply:
fixed via https://github.com/wso2/carbon-apimgt/pull/11786/commits/66bdca8d65de745ad762e0ded3b5e88c33b26016